packages/thirdweb/src/wallets/create-wallet.ts

```diff
@@ -466,8 +475,24 @@ export function inAppWallet(
       emitter.emit("disconnect", undefined);
     },
     switchChain: async (newChain) => {
-      // simply set the new chain
-      chain = newChain;
+      if (createOptions?.accountAbstraction && client && account) {
+        // if account abstraction is enabled, reconnect to smart account on the new chain
+        const { autoConnectInAppWallet } = await import(
+          "./in-app/core/wallet/index.js"
+        );
+        const [connectedAccount, connectedChain] = await autoConnectInAppWallet(
+          {
+            chain: newChain,
+            client,
+          },
+          createOptions,
+        );
+        account = connectedAccount;
+        chain = connectedChain;
+      } else {
+        // it not, simply set the new chain
```

Comment:
The absolute tiniest of nits but there's a typo in this comment

Reply:
uuuuugh can't unsee xD